src/coins.cpp

```diff
@@ -443,6 +443,18 @@ double CCoinsViewCache::GetPriority(const CTransaction &tx, int nHeight) const
             dResult += coins->vout[txin.prevout.n].nValue * (nHeight-coins->nHeight);
         }
     }
+
+    // Joinsplits do not reveal any information about a note's amount or age.
+    // We need to ensure priority increases the longer a tx remains in the mempool, so we don't
+    // generate random values which would change on each invocation of this method.
+    // Instead, we use the anchor as a seed to generate the note value deterministically.
+    // Currently, we say that each JoinSplit has input notes of between 1 to 2 ZEC,
+    // and that the age of each note is half the age of the blockchain.
```

Comment:
Since priority is neither a consensus rule nor a network protocol issue, miners and nodes are free to calculate priority any which way they want.  For example, a miner could randomly assign a priority value to all transactions, or as a social good, set z transactions to the maximum priority.  A transaction creator would have to be sure that miners are running a certain algorithm before expending resources trying to gain an advantage.  Right now, it's a reasonable assumption that most miners are using the reference client, unmodified.  To eliminate gamification, we could use the txid hash instead of the anchor or just pick a very big magic number.